llvm/lib/Target/RISCV/RISCVInstrInfoVVLPatterns.td

```diff
@@ -2570,6 +2511,16 @@ foreach fvti = AllFloatVectors in {
                  fvti.RegClass:$merge, fvti.RegClass:$rs2, fvti.RegClass:$rs1, (fvti.Mask V0),
                  GPR:$vl, fvti.Log2SEW)>;
 
+  def : Pat<(fvti.Vector (riscv_vmerge_vl (fvti.Mask V0),
```

Comment:
We had this pattern for vselect_vl but not vmerge_vl. Add it for vmerge_vl avoids a regression.